lib/ddtrace/tracer.rb

```diff
@@ -383,9 +402,7 @@ def subscribe_on_error(span_op, error_handler)
     # Records the span (& its context)
     def record_span(span_op)
       begin
-        span_op.service ||= default_service
```

Comment:
Is setting the `default_service` as a fallback after when the span is ready to be flushed done elsewhere now?

Reply:
Hmmmm....

So the answer is mostly yes, which is why I removed this. See `Tracer#build_span_options`: it decides and specifies the `service` name before it builds the `SpanOperation`. A `Tracer` spec that tests for `default_service` confirms this behavior is working correctly.

The "maybe no" part of this is that its possible for a `SpanOperation` to be manually created, not have a `service` specified, then flushed through a `Tracer#record`. In reality, this is never done in practice, or recommended. The `Tracer` only knows how to flush `Context` objects anyways, not individual `Span`s or `SpanOperation`s, so it's basically impossible.

All that said, I think this should be a safe change.